pkgs/applications/networking/instant-messengers/deltachat-desktop/default.nix

```diff
@@ -0,0 +1,132 @@
+{ lib
+, coreutils
+, electron
+, esbuild
+, fetchFromGitHub
+, fetchgit
+, fetchpatch
+, fetchurl
+, makeDesktopItem
+, makeWrapper
+, nix-gitignore
+, nodejs
+, nodePackages
+, pkgs
+, python2
+, runCommand
+, stdenv
+, writeTextFile
+}:
+
+let
+  nodeEnv = import ../../../../development/node-packages/node-env.nix {
+    inherit lib stdenv nodejs python2 pkgs runCommand writeTextFile;
+    libtool = if stdenv.isDarwin then pkgs.darwin.cctools else null;
+  };
+
+  # obtained using `node2nix --development`
+  # `--lock` does not work due to https://github.com/svanderburg/node2nix/issues/247
+  deltachatNodePackages = import ./node-packages.nix {
+    inherit nodeEnv fetchurl fetchgit nix-gitignore stdenv lib;
+  };
+
+  version = "1.20.3";
+  src = fetchFromGitHub {
+    owner = "deltachat";
+    repo = "deltachat-desktop";
+    # v1.20.3 + https://github.com/deltachat/deltachat-desktop/pull/2309
+    rev = "fcb637feb6e550f1c3761c696aa343a8aa359742";
+    sha256 = "0aad3qaq3sv9ba9p3326smbkgg26q8gka7ica03l3lfqab57a2ys";
+  };
+
+  esbuild' = esbuild.overrideAttrs (old: rec {
+    version = "0.8.57";
+    src = fetchFromGitHub {
+      owner = "netlify";
+      repo = "esbuild";
+      rev = "v${version}";
+      sha256 = "0axjyd3zfzkf0f73iyy0isacq6msrn1wq4cdf6f58gmw518q9vny";
+    };
+  });
+
+  inherit (deltachatNodePackages.shell.override (old: {
+    inherit src;
+    nativeBuildInputs = with nodePackages; [
+      node-gyp-build
+      typescript
+    ];
+    ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+    ESBUILD_BINARY_PATH = "${esbuild'}/bin/esbuild";
+  })) nodeDependencies;
+in stdenv.mkDerivation rec {
+  pname = "deltachat-desktop";
+  inherit version src;
+
+  nativeBuildInputs = [
+    makeWrapper
+    nodejs
+  ];
+
+  VERSION_INFO_GIT_REF = src.rev;
+
+  buildPhase = ''
+    runHook preBuild
+
+    export HOME=$TEMP
+
+    cp -r ${nodeDependencies}/lib/node_modules .
+    chmod -R u+w node_modules
+    # FIXME: we should not rely on prebuilt libdeltachat
+    # rm -r node_modules/deltachat-node/prebuilds
```

Comment:
I will work on this next but it will require upstream changes. See https://github.com/deltachat/deltachat-node/issues/509.
The package works as is so we can merge before I'm done with that.